### PR TITLE
Convert to EPP templates and add Puppet 4+ data types to parameters

### PR DESCRIPTION
**Pull request by alexjfisher**
* Convert erb templates to epp
* Add puppet 4+ data types

@jovandeginste I cherry-picked your commit from https://github.com/pdxcat/puppet-module-nrpe/pull/67, tightened a few things up (eg. using stdlib types), and made some of the breaking changes we'd been discussing.

### CODE REVIEW

File: functions/ssl_logging.pp [hunk 1]
```diff
@@ -0,0 +1,24 @@
+function nrpe::ssl_logging(
+  Boolean $ssl_log_startup_params,
+  Boolean $ssl_log_remote_ip,
+  Boolean $ssl_log_protocol_version,
+  Boolean $ssl_log_cipher,
+  Boolean $ssl_log_client_cert,
+  Boolean $ssl_log_client_cert_details,
+) >> String {
```
Comment (bastelfreak) [inline functions/ssl_logging.pp]: very interesting syntax

Comment (bastelfreak) [reply functions/ssl_logging.pp]: could you add a comment to the function and describe the purpose?

File: metadata.json [hunk 2]
```diff
@@ -8,6 +8,14 @@
   "project_page": "https://github.com/voxpupuli/puppet-nrpe",
   "issues_url": "https://github.com/voxpupuli/puppet-nrpe/issues",
   "dependencies": [
+    {
+      "name": "puppetlabs/stdlib",
+      "version_requirement": ">= 4.25.1 < 6.0.0"
+    },
+    {
+      "name": "puppetlabs/concat",
+      "version_requirement": ">= 3.0.0 < 6.0.0"
```
Comment (bastelfreak) [inline metadata.json]: does version 3 actually work? If I add a dependency I always set the lower version boundary to the current major release.

Comment (alexjfisher) [reply metadata.json]: yes, probably.  The current major release is only a couple of months old.

File: functions/ssl_logging.pp (hunk 1 above)
Comment (alexjfisher) [reply functions/ssl_logging.pp]: Done.  I could have built the string inside the `epp`, but decided to pull it into a separate function.

Comment (ekohl) [reply functions/ssl_logging.pp]: I think you can use regular rspec function tests on them as well.

Comment (alexjfisher) [reply functions/ssl_logging.pp]: Yes, it seemed easier to test it that way.

**Comment by jovandeginste** [comment]
@alexjfisher cool; should we close pdxcat#67 now?

**Pull request merged**